packages/kbn-observable/src/operators/__tests__/mergeMap.test.ts

```diff
@@ -0,0 +1,335 @@
+import { Observable } from '../../Observable';
+import { k$ } from '../../k$';
+import { BehaviorSubject } from '../../BehaviorSubject';
+import { mergeMap, delay } from '../';
+import { $of, $concat } from '../../factories';
+
+it('should map values to constant resolved promises and merge', done => {
+  expect.assertions(1);
+
+  const source = Observable.from([4, 3, 2, 1]);
+  const results: any[] = [];
+
+  k$(source)(mergeMap(() => Promise.resolve(42))).subscribe({
+    next(x) {
+      results.push(x);
+    },
+    error(err) {
+      done(err);
+    },
+    complete() {
+      expect(results).toEqual([42, 42, 42, 42]);
+      done();
+    }
+  });
+});
+
+it('should map values to constant rejected promises and merge', done => {
+  expect.assertions(1);
+
+  const source = Observable.from([4, 3, 2, 1]);
+
+  k$(source)(mergeMap(() => Promise.reject(42))).subscribe({
+    next(val) {
+      done(new Error(`next handled unexpectedly called with value: ${val}`));
+    },
+    error(err) {
+      expect(err).toBe(42);
+      done();
+    },
+    complete() {
+      done(new Error(`complete handled unexpectedly called`));
+    }
+  });
+});
+
+it('should map values to resolved promises and merge', done => {
+  expect.assertions(1);
+
+  const source = Observable.from([4, 3, 2, 1]);
+  const project = (value: number, index: number) =>
+    Promise.resolve(value + index);
+
+  const results: number[] = [];
+  k$(source)(mergeMap(project)).subscribe({
+    next(x) {
+      results.push(x);
+    },
+    error(err) {
+      done(err);
+    },
+    complete() {
+      expect(results).toEqual([4, 4, 4, 4]);
```

Comment:
__optional nit:__ this assert is tricky and a bit non-obvious :) Maybe you can map values into something that doesn't look like a first number in the source observable?